server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/CleanUpBulkImport.java

```diff
@@ -108,13 +108,12 @@ private static void removeBulkLoadEntries(Ample ample, TableId tableId, FateId f
           var tabletsMutator = ample.conditionallyMutateTablets()) {
 
         for (var tablet : tablets) {
-          // ELASTICITY_TODO DEFERRED - ISSUE 4044
-          if (tablet.getLoaded().values().stream().anyMatch(l -> l == fateId.getTid())) {
+          if (tablet.getLoaded().values().stream()
+              .anyMatch(loadedFateId -> loadedFateId.equals(fateId))) {
             var tabletMutator =
                 tabletsMutator.mutateTablet(tablet.getExtent()).requireAbsentOperation();
-            tablet.getLoaded().entrySet().stream()
-                .filter(entry -> entry.getValue() == fateId.getTid()).map(Map.Entry::getKey)
-                .forEach(tabletMutator::deleteBulkFile);
+            tablet.getLoaded().entrySet().stream().filter(entry -> entry.getValue().equals(fateId))
```

Comment:
Hard to tell from looking at this diff, but I assume the types line up?  Meaning `entry.getValue()` is a FateId type.

Reply:
Yes, types line up. TabletMetadata.getLoaded() (tablet.getLoaded()) returns Map<StoredTabletFile, FateId>

Reply:
Thanks